### ISSUE TITLE
Escaping of ' is missing in the Developer tools, leading to wrong syntax highlighting

### ISSUE THREAD

**Issue by RubenKelevra** [home-assistant/frontend#11576]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

This minor bug affects the proper syntax highlighting of entries in Set State's "State attributes"-view.

If a friendly name contains a ' (this friendly name was created by the official HA android app) it isn't properly escaped and thus will mess up the syntax highlighting in this view.

![Escaping missing on set_status](https://user-images.githubusercontent.com/614929/152681060-4fa4d9fd-54b4-47b7-8e12-757807c6450a.png)

### Describe the behavior you expected

Escaping with "" at the beginning/end of the string to get proper syntax highlighting 

### Steps to reproduce the issue

1. open developer tools
2. select an entry with a friendly name containing an '
3. observe wrong syntax highlighting

### What version of Home Assistant Core has the issue?

core-2022.2.2

### What was the last working version of Home Assistant Core?

unknown

### In which browser are you experiencing the issue with?

Chromium: 98.0.4758.87 / Firefox 95.0.2

### Which operating system are you using to run this browser?

Windows 11

**Comment by bdr99** [comment]
This is still an issue in 2022.5.

**Comment by bdr99** [comment]
Still an issue in 2022.8.

**Comment by bdr99** [comment]
This should not have been closed, the problem still exists in 2022.11. Can this issue be reopened?

**Comment by bdr99** [comment]
Still an issue in 2024.2.1.

**Comment by bdr99** [comment]
Yeah, I also noticed that their examples doesn't have this problem. That's why I think it might be an HA bug.

For future reference, this is how I tested it:

1. Go to https://codemirror.net/try/
2. Enter this code on the left side:
```javascript
import { basicSetup, EditorView } from "codemirror";
import { yaml } from "@codemirror/lang-yaml";

new EditorView({
  doc: "key: value'value'",
  extensions: [basicSetup, yaml()],
  parent: document.body
});
```
3. Click "Run"
4. Observe result (syntax highlighting works correctly):
![image](https://github.com/home-assistant/frontend/assets/2292715/c7e7f4d9-9162-498b-84aa-ece39f074cbc)

**Comment by silamon** [comment]
We're still using the legacy yaml mode: https://codemirror.net/5/mode/yaml/ where the bug is present

**Comment by bdr99** [comment]
Still an issue in 2024.8.2.

**Comment by bdr99** [comment]
Still an issue in 2025.3.3.

**Comment by prashker** [comment]
**Core
2025.3.3
Frontend
20250306.0**

Similar (related?) issue, including {# jinja comments #} in a value_template is not properly "comment" colorized when represented as YAML.
**This works properly (in my case this is an automation condition that does evaluate true/false as expected) but looks weird.** When accessed in visual editor, it is colored correctly.

![Image](https://github.com/user-attachments/assets/4bd2fbb8-980f-4c83-a392-1ebb138e2366)

![Image](https://github.com/user-attachments/assets/c95ff4f1-b633-42e3-b781-ac8a3078ca83)


YAML for reference.

```
condition: template
value_template: >-
  {# 20 Days Since End of Winter Based off March 20 of each year (see if this
  needs to change?) #} 

  {{ ((as_timestamp(now()) - as_timestamp(now().year ~ '-03-20')) / 86400 ) |
  int < 20 }}
```

**Comment by bdr99** [comment]
Still an issue in 2025.6.1.

**Comment by bdr99** [comment]
Still an issue in 2025.9.3.
